### PR TITLE
feat: Deduplicate share modal and members page

### PR DESCRIPTION
**Pull request by yaldram**
## Description
- Deduplicate the `FormDialog` component.
- All the Share application & Invite Dialog de-duplicated.
- Deduplicate Members page

Fixes [#22312](https://github.com/appsmithorg/appsmith/issues/22312) [#22961](https://github.com/appsmithorg/appsmith/issues/22961) [#19390](https://github.com/appsmithorg/appsmith/issues/19390)

Media
Old (by Arsalan) - https://www.loom.com/share/2f4238e412b346cf8d616ec08ca9e8e5
New (by Ankita) - https://cjrc9tc9nt.vmaker.com/record/UPBLs1sHPP2Fuwdl

## Type of change
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)

## How Has This Been Tested?
- Manual
- Jest

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/ce/pages/Applications/EmbedSnippetTab.tsx [hunk 1]
```diff
@@ -84,154 +34,141 @@ export function EmbedSnippetTab({
...
+
+      <EmbedCodeSnippet
+        isAppSettings={false}
+        snippet={embedSnippet.appViewEndPoint}
+      />
+
+      <BottomWrapper className={`flex justify-end pt-5`}>
+        <Link
+          className="flex gap-1 items-center self-end"
+          data-cy="preview-embed"
```
Comment (albinAppsmith) [inline app/client/src/ce/pages/Applications/EmbedSnippetTab.tsx:110]: Please rename this to data-testid

File: app/client/src/ce/pages/Applications/EmbedSnippetTab.tsx [hunk 2]
```diff
@@ -84,154 +34,141 @@ export function EmbedSnippetTab({
...
-    </EmbedSnippetContainer>
-  ) : (
-    <PrivateEmbeddingContent
-      canMakeAppPublic={canShareWithPublic}
-      changeTab={changeTab}
-      isAppSettings={isAppSettings}
-    />
+
+      <Switch
+        data-cy={"show-navigation-bar-toggle"}
```
Comment (albinAppsmith) [inline app/client/src/ce/pages/Applications/EmbedSnippetTab.tsx:158]: data-testid

File: app/client/src/pages/Editor/EditorHeader.tsx [hunk 3]
```diff
@@ -473,21 +433,65 @@ export function EditorHeader(props: EditorHeaderProps) {
...
+                  kind="tertiary"
+                  onClick={() => setShowModal(true)}
+                  size="md"
+                  startIcon="share-line"
+                >
+                  {createMessage(EDITOR_HEADER.share)}
+                </Button>
+              </Tooltip>
             )}
+            <Modal
```
Comment (rohitagarwal88) [inline app/client/src/pages/Editor/EditorHeader.tsx:459]: @ankitakinger : Shouldn't this modal be common for share?

Comment (ankitakinger) [reply app/client/src/pages/Editor/EditorHeader.tsx:459]: Since we have tabs here and nowhere else, this has been kept separate intentionally.

**Review by rohitagarwal88 — APPROVED** [review]
Looks good!

**Pull request merged**